build/build.sh

```diff
@@ -49,6 +49,7 @@ echo "Building targets for ${ARCH}, generated targets in ${TARGETS_DIR} director
 
 echo "Building ${PKG}/cmd/nginx"
 
+git config --add safe.directory /go/src/k8s.io/ingress-nginx
```

Comment:
if this doesn't runs inside a container, will it error?

Reply:
To me, `build.sh` runs only inside a container.
Then I searched occurrences of `build.sh` and I see it either in a Dockerfile or  in the Make target `build` (and Make target build is `run-in-docker.sh`) .

<img width="810" alt="image" src="https://user-images.githubusercontent.com/5085914/180047885-dcc7e57d-5e93-440b-8ae1-2000cfe8391c.png">


My thought is that as soon as a new environment(container) spawns for build, it all depends not the git version, for this problem (safe.directory) to occur or not. Hence this change from me.

If this problem is not happening for anyone else, then it will help to know details as to what they were doing and how they executed `make build` (as in any info on the git version inside that spawned container).